rewriter/RBSSignatures.cc

```diff
@@ -0,0 +1,232 @@
+        string_view preDefinition = sourceCode.substr(0, sourceCode.rfind('\n', beginIndex));
+
+        // Get all the lines before it
+        std::vector<string_view> all_lines = absl::StrSplit(preDefinition, '\n');
+
+        // We compute the current position in the source so we know the location of each comment
+        uint32_t index = beginIndex;
+
+        // Iterate from the last line, to the first line
+        for (auto it = all_lines.rbegin(); it != all_lines.rend(); it++) {
+            index -= it->size();
+            index -= 1;
+
+            string_view line = absl::StripAsciiWhitespace(*it);
+
+            // Short circuit when line is empty
+            if (line.empty()) {
+                break;
+            }
+
+            // Handle single-line sig block
+            else if (absl::StartsWith(line, "sig")) {
+                // Do nothing for a one-line sig block
+                // TODO: Handle single-line sig blocks
+            }
+
+            // Handle multi-line sig block
+            else if (absl::StartsWith(line, "end")) {
+                // ASSUMPTION: We either hit the start of file, a `sig do`/`sig(:final) do` or an `end`
+                // TODO: Handle multi-line sig blocks
+                it++;
+                while (
+                    // SOF
+                    it != all_lines.rend()
+                    // Start of sig block
+                    && !(absl::StartsWith(absl::StripAsciiWhitespace(*it), "sig do") ||
+                         absl::StartsWith(absl::StripAsciiWhitespace(*it), "sig(:final) do"))
+                    // Invalid end keyword
+                    && !absl::StartsWith(absl::StripAsciiWhitespace(*it), "end")) {
+                    it++;
+                };
+
+                // We have either
+                // 1) Reached the start of the file
+                // 2) Found a `sig do`
+                // 3) Found an invalid end keyword
+                if (it == all_lines.rend() || absl::StartsWith(absl::StripAsciiWhitespace(*it), "end")) {
+                    break;
+                }
+
+                // Reached a sig block.
+                line = absl::StripAsciiWhitespace(*it);
+                ENFORCE(absl::StartsWith(line, "sig do") || absl::StartsWith(line, "sig(:final) do"));
+
+                // Stop looking if this is a single-line block e.g `sig do; <block>; end`
+                if ((absl::StartsWith(line, "sig do;") || absl::StartsWith(line, "sig(:final) do;")) &&
+                    absl::EndsWith(line, "end")) {
+                    break;
+                }
+
+                // Else, this is a valid sig block. Move on to any possible documentation.
+            }
+
+            // Handle a RBS sig annotation `#: SomeRBS`
+            else if (absl::StartsWith(line, "#:")) {
+                // Account for whitespace before comment e.g
+                // # abc -> "abc"
+                // #abc -> "abc"
```

Comment:
This example comment should probably use RBS syntax, e.g. `//#: abc -> "abc"`.

The whitespace is also...after the comment?  Maybe the comment should say "before the annotation"?